sound/soc/soc-topology.c

```diff
@@ -2377,8 +2377,11 @@ static int soc_tplg_link_elems_load(struct soc_tplg *tplg,
 		}
 
 		ret = soc_tplg_link_config(tplg, _link);
-		if (ret < 0)
+		if (ret < 0) {
+			if (!abi_match)
+				kfree(_link);
```

Comment:
Reduce indentation level by using a goto?

Reply:
@paulstelian97 I could do that. But that check is done inside a loop going through all the DAI links. For each such DAI link if an ABI mismatch is found, an ABI-compliant copy is created in the beggining of the iteration and then it is kfree()ed at the end. Seems a bit odd breaking the loop with a goto and deallocate copy outside.

Reply:
Fair, I'd wait on other reviewers to see if they agree with me though.